### PR TITLE
Non-dimensionalization with zref, ustar and pretty printing for issue 59

### PR DESCRIPTION
**Pull request by alizma**
Following our Friday meeting to properly non-dimensionalize, this PR combines those corrections with new pretty printing for calibrated parameters. The pretty printing looks like this: 

image

The ``.calibrate()`` method also now returns a dictionary of values with keys ``L``, ``Gamma``, ``sigma`` with the corresponding fitted values. Together, these additions should be enough to finish #59. 

With regards to the non-dimensionalization, I'm still concerned with further updates to 1) penalty terms in the loss function, which now require additional divisions by ``zref`` and 2) the data interpolation in the data generator, which requires a multiplication by ``zref``... I was hoping to confirm that I propagated these changes correctly. @brendankeith is there any chance you could take a look, please? The relevant functions are the penalty loss functions (e.g. second order [here](https://github.com/METHODS-Group/DRDMannTurb/blob/19c36d8a11f6688f941e3c4972a7ad0c51c3a9ba/drdmannturb/spectra_fitting/loss_functions.py#L129-L132)) in ``LossAggregator`` and the ``DataType.AUTO`` branch in the data generator ([here](https://github.com/METHODS-Group/DRDMannTurb/blob/19c36d8a11f6688f941e3c4972a7ad0c51c3a9ba/drdmannturb/spectra_fitting/data_generator.py#L114-L164)).

### CODE REVIEW

**Comment by alizma** [comment]
This PR now also includes basic versions of tests listed in #44.

File: drdmannturb/spectra_fitting/calibration.py [hunk 1]
```diff
@@ -705,10 +742,14 @@ def plot(
...
-        kF_model_vals = model_vals if model_vals is not None else self.OPS(k1_data_pts)
+        kF_model_vals = (
+            model_vals
+            if model_vals is not None
+            else self.OPS(k1_data_pts) / self.phys_params.ustar**2.0
+        )
 
         kF_model_vals = kF_model_vals.cpu().detach()
-        kF_data_vals = kF_data_vals.cpu().detach()
+        kF_data_vals = kF_data_vals.cpu().detach() / self.phys_params.ustar**2
```
Comment (gdeskos) [inline drdmannturb/spectra_fitting/calibration.py:752]: Nice!

File: drdmannturb/spectra_fitting/data_generator.py [hunk 2]
```diff
@@ -326,11 +330,12 @@ def eval_Kaimal(self, k1: float) -> torch.Tensor:
...
-        F[1, 1] = 17 * n / (1 + 9.5 * n) ** (5 / 3)
-        F[2, 2] = 2.1 * n / (1 + 5.3 * n ** (5 / 3))
-        F[0, 2] = -12 * n / (1 + 9.6 * n) ** (7.0 / 3.0)
-        return F
+        F[0, 0] = 52.5 * n / (1 + 33 * n) ** (5 / 3)
+        F[1, 1] = 8.5 * n / (1 + 9.5 * n) ** (5 / 3)
+        F[2, 2] = 1.05 * n / (1 + 5.3 * n ** (5 / 3))
+        F[0, 2] = -7 * n / (1 + 9.6 * n) ** (12.0 / 5.0)
+
+        return F * self.ustar**2
```
Comment (gdeskos) [inline drdmannturb/spectra_fitting/data_generator.py:338]: These numbers are the correct ones! Thanks for catching this!

File: drdmannturb/spectra_fitting/loss_functions.py [hunk 3]
```diff
@@ -121,8 +126,10 @@ def Pen2ndOrder(self, y: torch.Tensor, epoch: int) -> torch.Tensor:
         logy = torch.log(torch.abs(y))
         d2logy = torch.diff(torch.diff(logy, dim=-1) / self.h1, dim=-1) / self.h2
 
-        pen2ndorder_loss = self.params.alpha_pen2 * torch.mean(
-            torch.relu(d2logy).square()
+        pen2ndorder_loss = (
+            self.params.alpha_pen2
+            * torch.mean(torch.relu(d2logy).square())
+            / self.zref**2
```
Comment (gdeskos) [inline drdmannturb/spectra_fitting/loss_functions.py:132]: Not sure why we divide by zref^2 here

File: examples/plot_mann_fit.py [hunk 4]
```diff
@@ -55,7 +55,7 @@
 
 print(f"Physical Parameters: {L,Gamma,sigma}")
 
-domain = torch.logspace(-1, 2, 20)
+k1 = torch.logspace(-1, 2, 20) / zref
```
Comment (gdeskos) [inline examples/plot_mann_fit.py:58]: Nice!

**Review by gdeskos — APPROVED** [review]

**Comment by brendankeith** [comment]
> Following our Friday meeting to properly non-dimensionalize, this PR combines those corrections with new pretty printing for calibrated parameters. The pretty printing looks like this:
> 
> image
> 
> The `.calibrate()` method also now returns a dictionary of values with keys `L`, `Gamma`, `sigma` with the corresponding fitted values. Together, these additions should be enough to finish #59.
> 
> With regards to the non-dimensionalization, I'm still concerned with further updates to 1) penalty terms in the loss function, which now require additional divisions by `zref` and 2) the data interpolation in the data generator, which requires a multiplication by `zref`... I was hoping to confirm that I propagated these changes correctly. @brendankeith is there any chance you could take a look, please? The relevant functions are the penalty loss functions (e.g. second order [here](https://github.com/METHODS-Group/DRDMannTurb/blob/19c36d8a11f6688f941e3c4972a7ad0c51c3a9ba/drdmannturb/spectra_fitting/loss_functions.py#L129-L132)) in `LossAggregator` and the `DataType.AUTO` branch in the data generator ([here](https://github.com/METHODS-Group/DRDMannTurb/blob/19c36d8a11f6688f941e3c4972a7ad0c51c3a9ba/drdmannturb/spectra_fitting/data_generator.py#L114-L164)).

The fits are independent of `ustar`, but I am confused about why even when we run `plot_mann_fit.py`, we get different initial losses for different values of `zref`.

This seems like another scaling issue to me

**Comment by brendankeith** [comment]
> DataType.AUTO

The data interpolation part looks good to me, but I wasn't able to review the entire code closely. I suggest we try to meet in person next week or wait until the following week when my grant proposal is over.

**Comment by alizma** [comment]
> The fits are independent of `ustar`, but I am confused about why even when we run `plot_mann_fit.py`, we get different initial losses for different values of `zref`.
> 
> This seems like another scaling issue to me

I'm not sure why this is happening, but here are a few comments: this loss evaluation involves only the MSE term in the loss, which is computed between the output of the data generator and the initial output from the OPS.  My suspicion is the issue is in the scaling for the OPS, since ``zref`` is not involved anywhere in its evaluation and just takes in the ``k1`` passed during the ``.calibrate()`` call.

**Comment by alizma** [comment]
Since the initial PR was opened, several additions were made that close #59, #57, #55, and #44. Please see the updated documentation re-hosted at https://methods-group.github.io/DRDMannTurb/
